dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java

```diff
@@ -246,4 +259,30 @@ private Tool findToolByIdAndCheckToolAndUser(Long toolId, User user) {
         checkUser(user, tool);
         return tool;
     }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("{containerId}/tags/{tagId}/sourcefiles")
+    @Operation(operationId = "getTagsSourcefiles", description = "Retrieve sourcefiles for a container's version", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public SortedSet<SourceFile> getTagsSourceFiles(@ApiParam(hidden = true) @Auth Optional<User> user,
+            @Parameter(name = "containerId", description = "Container to retrieve the version from", required = true, in = ParameterIn.PATH) @PathParam("containerId") Long containerId,
+            @Parameter(name = "tagId", description = "Tag to retrieve the sourcefiles from", required = true, in = ParameterIn.PATH) @PathParam("tagId") Long tagId,
+            @Parameter(name = "fileTypes", description = "List of file types to filter sourcefiles by") @QueryParam("fileType") List<DescriptorLanguage.FileType> fileTypes) {
+        Tool tool = toolDAO.findById(containerId);
+        checkEntry(tool);
+        checkOptionalAuthRead(user, tool);
+
+        Version version = tagDAO.findVersionInEntry(containerId, tagId);
```

Comment:
Probably don't even need to change visibility since TagDAO is a VersionDAO.

Reply:
it's so that the other endpoint could share the same code

Reply:
I made a new query because it was asked to verify that the version belongs to the entry requested, and that it'd be the the quicker option, but I'll use versionDAO instead.